File: ruby/tabulation/fib.rb
def fib(n)
    table = Array.new(n+1, 0)
    table[1] = 1
    n.times do |i|
        if i+1 <= n
        table[i+1] += table[i]
        end
        if i+2 <= n
        table[i+2] += table[i]
        end
    end
    table[n]
end

p fib(7)

File: javascript/tabulation/fib.js
const fib = n => {
    const table = Array(n + 1).fill(0);
    table[1] = 1;
    for (let i = 0; i <= n; i++ ) {
        console.log(i)
        console.log(table);
        table[i+1] += table[i]
        table[i+2] += table[i]
    }

    return table[n];
};


console.log(fib(6));